### PR TITLE
Truncate messages based on bytes, not unicode characters

### PR DESCRIPTION
**Pull request by tomkins**
We've been getting this message quite a bit in one of our apps:

```
site-packages/watchtower/__init__.py:339: WatchtowerWarning: Failed to deliver logs: An error occurred (InvalidParameterException) when calling the PutLogEvents operation: Upload too large: 1048681 bytes exceeds limit of 1048576
```

Some of the messages we're sending contain unicode characters, and we're sending a **lot** of messages in the batches. However as the message size calculation is counting the length of strings, instead of the number of bytes - we're getting warnings originating from boto3.

Temporary workaround is to drop the message/batch size - however the better fix is to change the size checking and truncation to look at the bytes of the messages.

### CODE REVIEW

File: watchtower/__init__.py [hunk 1]
```diff
@@ -45,6 +45,20 @@ def _boto_filter(record):
     return True
 
 
+def _size(msg):
```
Comment (kislyuk) [inline watchtower/__init__.py:48]: Can you move _size to be a method of CloudWatchLogHandler to simplify customization by subclassing?

File: watchtower/__init__.py [hunk 2]
```diff
@@ -45,6 +45,20 @@ def _boto_filter(record):
...
+    # Calculate the byte size of a message - accounting for unicode, and extra payload size
+    return (
+        len(msg["message"].encode("utf-8")) if isinstance(msg, dict) else 1
+    ) + CloudWatchLogHandler.EXTRA_MSG_PAYLOAD_SIZE
+
+
+def _truncate(msg, max_size):
```
Comment (kislyuk) [inline watchtower/__init__.py:55]: Same here - Can you move _truncate to be a method of CloudWatchLogHandler to simplify customization by subclassing?

**Review by kislyuk — CHANGES_REQUESTED** [review]
Thanks for your contribution and for chasing down the bug! I just have two minor suggestions, it looks great otherwise.

**Comment by tomkins** [comment]
Updated, and hopefully found an okay place to put them.

Kept them as `_` methods, but can change that back if needed.

**Comment by kislyuk** [comment]
Released in v3.0.1

**Pull request merged**